pkg/cluster/spec.go

```diff
@@ -142,22 +137,41 @@ func WithManagementCluster(cluster *types.Cluster) SpecOpt {
 	}
 }
 
-func NewSpec(clusterConfigPath string, cliVersion version.Info, opts ...SpecOpt) (*Spec, error) {
-	clusterConfig, err := eksav1alpha1.GetClusterConfig(clusterConfigPath)
-	if err != nil {
-		return nil, err
+func WithUserAgent(userAgent string) SpecOpt {
+	return func(s *Spec) {
+		s.userAgent = userAgent
 	}
+}
 
+func NewSpec(opts ...SpecOpt) *Spec {
 	s := &Spec{
 		releasesManifestURL: releasesManifestURL,
 		configFS:            configFS,
-		httpClient:          &http.Client{},
-		userAgent:           userAgent("cli", cliVersion.GitVersion),
+		userAgent:           userAgent("unknown", "unknown"),
 	}
+
 	for _, opt := range opts {
 		opt(s)
 	}
 
+	s.reader = s.newManifestReader()
+
+	return s
+}
+
+func newWithCliVersion(cliVersion version.Info, opts ...SpecOpt) *Spec {
+	opts = append(opts, WithUserAgent(userAgent("cli", cliVersion.GitVersion)))
```

Comment:
What about "eks-a-cli" here? We aren't using these user agents yet right?

Reply:
good point, but same, the `userAgent` will add that